addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -26,12 +26,16 @@ def _compute_is_forward_operator(self):
         self.filtered(lambda s: s.step_type == "create_lead_and_forward").is_forward_operator = True
 
     def _chatbot_crm_prepare_lead_values(self, discuss_channel, description):
+        partner = self.env.user.partner_id
+        team = self.crm_team_id
+        if partner.company_id and team.company_id and partner.company_id != team.company_id:
+            team = self.env["crm.team"]
```

Comment:
That's not the right move. Team is important, company is not.

Reply:
It's impossible to fix it differently. The only possible way to assign the partner (with a company) on the lead is to have the company match.

Reply:
After checking again we don't have many other ways to solve it.

Main issue: we can't assign a partner and a team which have different companies. I take that for granted as I don't think we will change multi-company rules here.

We can't remove the company from the partner.
We can't remove the company from the team.

We could not assign the partner and keep the team, but it feels very counter-intuitive to be able to link the lead to an existing partner and not do it, just to preserve the team.

So the best solution for me is to assign the partner and ignore the team in this very specific case.